src/devices/Tlc59711/Tlc59711.cs

```diff
@@ -0,0 +1,155 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.Device.Gpio;
+using System.Device.Spi;
+using System.IO;
+using System.Linq;
+using System.Text;
+using System.Text.RegularExpressions;
+using Iot.Device.Spi;
+
+namespace Iot.Device.Tlc59711
+{
+    /// <summary>
+    /// On-board LED on the device.
+    /// </summary>
+    public class Tlc59711 : IDisposable
+    {
+        private readonly ushort[] _pwmBuffer;
+        private readonly byte _numberOfDrivers;
+        private SpiDevice _spiDevice;
+        private byte _brightnessRed;
+        private byte _brightnessGreen;
+        private byte _brightnessBlue;
+        private byte[] _dataToWrite;
+
+        /// <summary>
+        /// Creates a new instance of the Tlc59711.
+        /// </summary>
+        public Tlc59711(byte numberOfDrivers, SpiDevice spiDevice)
+        {
+            _numberOfDrivers = numberOfDrivers;
+
+            _spiDevice = spiDevice ?? throw new ArgumentNullException(nameof(spiDevice));
+
+            // default 100% brigthness
+            _brightnessRed = _brightnessGreen = _brightnessBlue = 0x7f;
+
+            // 12 channels per driver
+            _pwmBuffer = new ushort[numberOfDrivers * 12];
+
+            _dataToWrite = new byte[_numberOfDrivers * 28];
+        }
+
+        /// <summary>
+        /// Disposes SpiDevice instances
+        /// </summary>
+        public void Dispose()
+        {
+            _spiDevice?.Dispose();
+            _spiDevice = null;
+        }
+
+        /// <summary>
+        /// Call this to set the PWM level for a channel.
+        /// </summary>
+        /// <param name="channel">Channel, 0-based</param>
+        /// <param name="pwm">PWM level (0 = minimum, 65535 = maximum)</param>
+        public void SetPWM(int channel, ushort pwm)
```

Comment:
I'd expect this work similarly to Pca9685, i.e.:
- this should allow creating PwmChannel
- global changing values is optional
- utilities for LED and similar I don't think they necessarily belong here (I'd rather have a separate PR with extension methods for all PwmChannels)

note the benefits of being able to construct PwmChannel:
- consistent APIs
- being able to pass it to other bindings which accept PwmChannel (i.e. ServoMotor)
- more in line with our on-going conversation of abstracting capabilities of the current board (i.e. board with TLC59711 with 1 driver would be equivalent of having 12 PwmChannels on board where having the hardware would be transparent for users)

Reply:
Well, to some degree, but the TLC59711 does not allow you to change the frequency, it will only allow you to control the DutyCycle. The TLC is a dedicated LED driver (which internally is using PWM).

Reply:
Would it make sense to just report the correct frequency but make setter a no-op? Most of the PWMs can be used to control LEDs. You can mention the limitation in the README.md

Reply:
I actually can't find the frequency this device is using, they have something called Enhanced Spectrum PWM, but they don't specify the frequency (maybe it's dynamic, I don't know). I don't disagree with you on using a standard API for (LED) control in the library, but I'm not sure it's a great fit with the PwmChannel IMO.

Reply:
I think the frequency is f_osc is between 7-12MHz (per spec it's typically 10Mhz) divided by PWM steps which is 65536 which gives around 150Hz. Probably easiest to check with the oscilloscope. If you think this is different enough it's ok not to implement PwmChannel but I suspect someone will do that in the future anyway given that ServoMotor won't work otherwise 😄 Please at least make the API as similar as possible to Pca9685

Reply:
Most LED drivers use a much higher PWM frequency than 150Hz, so I don't think that number is correct. If it's not known (or it may be variable and controlled by the driver, ie their "Enhanced Spectrum PWM" I don't think it makes sense to add something in the class here that may not be correct. Unfortunately I have moved and don't have my electronics workbench set up currently, so I can't verify with a scope. There's also a risk that the frequency changes with the PWM settings. I don't think the PwmChannel is a good fit.